zerver/actions/reminders.py

```diff
@@ -0,0 +1,122 @@
+import datetime
+import re
+
+from django.conf import settings
+from django.utils.translation import gettext as _
+
+from zerver.actions.message_send import check_message, internal_send_private_message
+from zerver.actions.scheduled_messages import do_schedule_messages
+from zerver.lib.addressee import Addressee
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.message import access_message
+from zerver.lib.url_encoding import huddle_narrow_url, personal_narrow_url, topic_narrow_url
+from zerver.models import (
+    Client,
+    Message,
+    ScheduledMessage,
+    Stream,
+    UserProfile,
+    get_display_recipient,
+    get_system_bot,
+)
+
+
+# This function is similar to one used in fenced_code.ts
+def get_unused_fence(content: str) -> str:
+    # Define the regular expression pattern to match ``` fences
+    fence_length_re = re.compile(r"^ {0,3}(`{3,})", re.MULTILINE)
+
+    # Initialize the length variable to 3, corresponding to default fence length
+    length = 3
+    matches = fence_length_re.findall(content)
+    for match in matches:
+        length = max(length, len(match) + 1)
+
+    return "`" * length
+
+
+def by_conversation_and_time_url(message: Message, current_user: UserProfile) -> str:
+    if message.is_stream_message():
+        stream_id = message.recipient.type_id
+        narrow_link = topic_narrow_url(
+            realm=message.sender.realm,
+            stream=Stream.objects.get(id=stream_id),
+            topic=message.topic_name(),
+        )
+    elif message.recipient.type == message.recipient.PERSONAL:
+        narrow_link = personal_narrow_url(
+            realm=message.sender.realm,
+            sender=message.sender,
+        )
+    else:
+        narrow_link = huddle_narrow_url(
+            user=current_user,
+            display_recipient=get_display_recipient(message.recipient),
+        )
+
+    return f"{narrow_link}/near/{message.id}"
```

Comment:
How does this differ from `near_message_url`?

Reply:
It expects message to be `Dict`, so it felt better do it this way.

Reply:
Hmm, there should be an easy function to dictify a message though that we can use...